network/ClientNetworking.h

```diff
@@ -53,10 +44,11 @@
 class ClientNetworking : public std::enable_shared_from_this<ClientNetworking> {
 public:
     /** The type of list returned by a call to DiscoverLANServers(). */
-    typedef std::vector<std::pair<boost::asio::ip::address, std::string> >  ServerList;
+    using ServerNames =  std::vector<std::string>;
```

Comment:
In most cases, and this specific case `using` and `typedef` are exactly the same.

tldr; We should replace all of our `typedefs` with `using` declarations.

`using` is more general than `typedef`.  

There is never case where employing `using` produces less functional or more verbose code than `typedef`.  

`using` can be templated and the compiler then knows that the templated thing is a type.  This results in less verbose code and better compiler errors than the typedef alternative.

For example create two templated types, one with `using` and one with `typedef`:
````
template<typename T> using UsingMappedThingType = std::map<T, Thing<T>>;
````
something similar with `typedef` involves nesting in a templated structure.
````
template<typename T> struct StructToHoldTypedef {
    typedef std::map<T, Thing<T>> TypdefMappedThingType;
};
```` 
When further utilizing those types in other templated things, the compiled knows that `UsingMapThingType<T>` is a type, but `StructToHoldTypedef<T>::TypedefMappedThingType` may be a type, member data or a member function, the compile does not know.    

In pre-C++11 code, you can add `typename` decorators to force the compiler to consider this a type, `typename StructToHoldTypedef<T>::TypedefMappedThingType`.  Sometimes adding/removing `typename` causes the compile to fail, because the member you used `typename` for is not actually a type.  In C++11 code, `UsingMappedThingType<T>` in unambiguously a type.

Changing all of the `typedef` statements to `using` declarations would make the code less verbose and more consistent. Keeping `typedef` in some places and `using` in locations where it avoids the use of `typename` adds unnecessary cognitive load.